ProcessLib/BalanceProcess/BalanceProcess.cpp

```diff
@@ -0,0 +1,68 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "BalanceProcess.h"
+
+#include <cassert>
+
+#include "ProcessLib/Utils/CreateLocalAssemblers.h"
+
+namespace ProcessLib
+{
+BalanceProcess::BalanceProcess(MeshLib::Mesh& mesh,
+                               std::size_t bulk_property_number_of_components)
+{
+    DBUG("Create local balance assemblers.");
+    // Populate the vector of local assemblers.
+    _local_assemblers.resize(mesh.getElements().size());
+
+    // Create single component dof in every of the mesh's nodes.
+    _mesh_subset_all_nodes.reset(
+        new MeshLib::MeshSubset(mesh, &mesh.getNodes()));
+
+    // Collect the mesh subsets in a vector.
+    std::vector<std::unique_ptr<MeshLib::MeshSubsets>> all_mesh_subsets;
+    std::generate_n(
+        std::back_inserter(all_mesh_subsets),
+        bulk_property_number_of_components,
+        [&]() {
+            return std::unique_ptr<MeshLib::MeshSubsets>{
+                new MeshLib::MeshSubsets{_mesh_subset_all_nodes.get()}};
+        });
+
+    // dof table maps mesh elements to local assemblers
```

Comment:
the comment seems wrong, because for each element there is one local assembler, no dof-table needed.